src/packaging/_manylinux.py

```diff
@@ -204,6 +204,11 @@ def _is_compatible(name: str, arch: str, version: _GLibCVersion) -> bool:
 
 
 def platform_tags(linux: str, arch: str) -> Iterator[str]:
+    if arch == "armv8l":
+        # armv8l wheels are not accepted on PyPI
+        # As long as we pass the the ABI check below,
+        # the armv7l wheels can be installed.
+        arch = "armv7l"
```

Comment:
This boils down to the fact that PyPI only accepts `armv7l` for now.
A more future proof way to do this would be to refactor `_have_compatible_abi` to return a list of architectures and yield multiple arch per manylinux version.

Reply:
Is there any reason PyPI has that restriction? Or should we ask PyPI to update to accept the newer architecture?

Reply:
For reference, the architectures accepted by PyPI are defined there : https://github.com/pypi/warehouse/blob/f2b3ed3724b064fab8621960fc8658a867b95e1e/warehouse/forklift/legacy.py#L95-L149

Those architectures were defined through the various PEPs for manylinux/musllinux, they are more or less equivalent to what distros offers, with, in this case, `armv7l`==`armhf`.

While there's nothing preventing PyPI from accepting other architectures, except maybe going through a new PEP, IMHO, it's not worth adding `armv8l` there.

I think the demand for 32-bit ARM wheels is most likely linked to armv7/armhf wheels availability for platforms like RaspberryPI or other embedded systems (e.g. OpenWRT based ones for musllinux).

When it comes to building/testing such `armv7l` wheels, there are multiple routes for CI:
- use such an armv7 system but then, you're on your own to manage a self-hosted CI node
- use qemu which is painfully slow
- use an aarch64 system in 32-bit mode, I only tested Travis CI AWS Graviton 2 nodes in 32-bit mode for now but it should also be applicable to at least CirrusCI, Circle CI both using AWS Graviton 2 and the future GitHub-Hosted aarch64 runners.

For that last route, being able to install `armv7l` is important, publishing `armv8l` wheels probably less so.